src/mlpack/methods/linear_regression/linear_regression.cpp

```diff
@@ -95,7 +100,7 @@ void LinearRegression::Predict(const arma::mat& points,
   {
     // We want to be sure we have the correct number of dimensions in the
     // dataset.
-    Log::Assert(points.n_rows == parameters.n_rows - 1);
+    util::CheckSameDimensionality(points, (size_t) (parameters.n_rows - 1), "LinearRegression::Predict()", "points");
```

Comment:
Let's say parameters.n_rows is 0, then this cast would lead it to capture UINT32_MAX since it circles back.

When this fails, user gets "mismatch size x and UNIT32_MAX". Which causes confusion. Is there any check which ensures parameters isn't an empty matrix? If not, might as well add.

Reply:
I think, we dont need to enforce any check because it wont be empty matrix.  
You can see the following lines of code in [Train()](https://github.com/mlpack/mlpack/pull/3164/files#diff-0a7b203752b2fc291fff7a2bf2648f855b35c542805745d83bc334cf6d41b688R45) to understand my point .

Reply:
Okay move ahead.  Make sure to append those changes as well.